src/mpl2/src/rtl_mp.cpp

```diff
@@ -188,6 +189,12 @@ bool rtl_macro_placer(const char* config_file,
   float outline_lx = 0.0;
   float outline_ly = 0.0;
 
+  assert(num_thread <= 0);
+  assert(num_run <= 0);
+  assert(num_level <= 0);
+  assert(num_worker <= 0);
+  assert(perturb_per_step <= 0);
```

Comment:
Don't use C assertions - use the logger if you have an error.  assert will either be (1) compiled out and do nothing or (2) exit the program - neither of which is good.

Reply:
Solved